api/src/OpenTelemetry/Environment.hs

```diff
@@ -0,0 +1,12 @@
+module OpenTelemetry.Environment (
+  isTrue,
+) where
+
+import qualified Data.CaseInsensitive as CI
+
+
+-- Have a look here for the specification.
+-- https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#boolean-value
+
+isTrue :: String -> Bool
+isTrue = ("true" ==) . CI.mk
```

Comment:
In #110, Ian mentions minimal dependencies as a project goal. Should we just do `Data.Char` here instead of pulling in `case-insensitive`? If we're only comparing to `[t, r, u, e]`, I can't imagine there are any gotchas that can affect us.

```suggestion
isTrue = ("true" ==) . map Char.toLower
```

I would also consider exporting something like `lookupBooleanEnv = fmap (maybe False isTrue) . lookupEnv` from this module. Since you've named it `OpenTelementry.Environment`, it seems a reasonable expectation to me that it would own the actual env interaction, and not just the `String` predicate. WDYT?

Reply:
I totally agree about reducing dependencies. I choose `case-insensitive` 'cause it's already in the list of the dependencies in multiple packages.
I originally wrote something like `lookupBooleanEnv`, But I would like to propose some small refactoring for collecting all the environment variables (eg: we could use `getEnvironment` instead of using `lookupEnv` so many times). Anyway, I can add a function like that, it would make `loadExporterEnvironmentVariables` a bit more readable.

Reply:
> I choose case-insensitive 'cause it's already in the list of the dependencies in multiple packages

Ah, fair point!

Reply:
To clarify my stance on dependencies a bit, the ones I want to keep as minimal as possible are specifically `hs-opentelemetry-api` and `hs-opentelemetry-sdk`. Especially `api`. The reasoning for that is that it's intended to be something that other library writers can use in their code as a dependency. It makes it far easier for them to upgrade their package bounds if there are fewer dependencies.

Integration packages I'm far less fussed about.

Reply:
I agree that in this instance I think we can go with the dumb version.

Reply:
done